src/pow.rs

```diff
@@ -0,0 +1,1892 @@
+            pub fn to_le_array_u32(self) -> [u32; 8] { self.0.to_le_array_u32() }
+
+            /// Converts `Self` to a big-endian array of `u16`s.
+            #[inline]
+            pub fn to_be_array_u16(self) -> [u16; 16] { self.0.to_be_array_u16() }
+
+            /// Converts `Self` to a little-endian array of `u16`s.
+            #[inline]
+            pub fn to_le_array_u16(self) -> [u16; 16] { self.0.to_le_array_u16() }
+
+            /// Converts `Self` to a big-endian array of `u8`s.
+            #[inline]
+            pub fn to_be_array_u8(self) -> [u8; 32] { self.0.to_be_array_u8() }
+
+            /// Converts `Self` to a little-endian array of `u8`s.
+            #[inline]
+            pub fn to_le_array_u8(self) -> [u8; 32] { self.0.to_le_array_u8() }
+
+            /// Converts `self` to a big-endian array of `u128`s.
+            #[inline]
+            pub fn to_be(self) -> [u128; 2] { self.to_be_array_u128() }
+
+            /// Converts `self` to a little-endian array of `u128`s.
+            #[inline]
+            pub fn to_le(self) -> [u128; 2] { self.to_le_array_u128() }
+
+            /// Converts `self` to a big-endian byte array.
+            #[inline]
+            pub fn to_be_bytes(self) -> [u8; 32] { self.to_be_array_u8() }
+
+            /// Converts `self` to a little-endian byte array.
+            #[inline]
+            pub fn to_le_bytes(self) -> [u8; 32] { self.to_le_array_u8() }
+        }
+
+        impl_from_unsigned_type!($ty, u8);
+        impl_from_unsigned_type!($ty, u16);
+        impl_from_unsigned_type!($ty, u32);
+        impl_from_unsigned_type!($ty, u64);
+        impl_from_unsigned_type!($ty, u128);
+
+        impl_try_from_signed_type!($ty, i8);
+        impl_try_from_signed_type!($ty, i16);
+        impl_try_from_signed_type!($ty, i32);
+        impl_try_from_signed_type!($ty, i64);
+        impl_try_from_signed_type!($ty, i128);
+
+        impl fmt::Display for $ty {
+            #[inline]
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::Display::fmt(&self.0, f) }
+        }
+
+        impl fmt::LowerHex for $ty {
+            #[inline]
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::LowerHex::fmt(&self.0, f) }
+        }
+
+        impl fmt::UpperHex for $ty {
+            #[inline]
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::UpperHex::fmt(&self.0, f) }
+        }
+
+        impl_div_mul!($ty, u8);
+        impl_div_mul!($ty, u16);
+        impl_div_mul!($ty, u32);
+        impl_div_mul!($ty, u64);
+
+        impl Shl<u32> for $ty {
+            type Output = Self;
+
+            #[inline]
+            fn shl(self, shift: u32) -> Self { Self(self.0 << shift) }
+        }
+
+        impl Shr<u32> for $ty {
+            type Output = Self;
+
+            #[inline]
+            fn shr(self, shift: u32) -> Self { Self(self.0 >> shift) }
+        }
+    };
+}
+
+/// A big-endian 256 bit integer representing work.
```

Comment:
"big-endian" is an implementation detail and doesn't make sense as part of the doccomment. (Same for `Target` below)

Reply:
Removed, thanks.